modules/core/include/opencv2/core/cuda/filters.hpp

```diff
@@ -95,6 +96,12 @@ namespace cv { namespace cuda { namespace device
 
             const int x1 = __float2int_rd(x);
             const int y1 = __float2int_rd(y);
+            if (x1 <= NPP_MIN_32S || x1 >= NPP_MAX_32S || y1 <= NPP_MIN_32S || y1 >= NPP_MAX_32S)
```

Comment:
`x < 0`
`y < 0`
`x >= src.width`
`y >= src.height`

Need to take border value somehow (not sure if it is properly defined for remap).

Reply:
src is an instance of the class BorderReader which decided how to handle values out of scope by itself depending on the type of border. Values greater than height and width are completely valid, we don't have to clamp them here.

Reply:
Please note, that condition checks for x1,y1 which have "int" type, instead of the original x,y which can underflow/overflow/NaN/Inf "int" range (return value from `__float2int_rd` is not something expected).